app/controllers/posts_controller.rb

```diff
@@ -1,6 +1,6 @@
 class PostsController < ApplicationController
 
-  before_action :authenticate_user!, only: [:new, :edit, :update, :destroy]
+  skip_before_action :authenticate_user!, only: [:index]
```

Comment:
詳細を見る（show）のもログインしなくていいはずなので、show足しましょうー！
```suggestion
  skip_before_action :authenticate_user!, only: [:index, :show]
```